src/app/app.ts

```diff
@@ -120,11 +122,11 @@ class AppImpl implements App {
         const adminClient = newMMClient(this.context).asActingUser();
 
         // add bot to team and channel
-        const botUserID = this.context.bot_user_id;
-        const addToTeamReq = adminClient.addToTeam(this.context.team_id, botUserID);
+        const botUserID = this.context.bot_user_id as string;
+        const addToTeamReq = adminClient.addToTeam(this.context.team_id as string, botUserID);
         await tryPromiseWithMessage(addToTeamReq, 'Failed to add bot to team');
 
-        const addToChannelReq = adminClient.addToChannel(botUserID, this.context.channel_id);
+        const addToChannelReq = adminClient.addToChannel(botUserID, this.context.channel_id as string);
```

Comment:
Why are these `as`'s needed? In which cases are the values `undefined`?

Reply:
@hanzei, the values require the `as` type casting when the type field is optional.  https://github.com/mattermost/mattermost-plugin-apps/blob/master/apps/context.go#L27

Reply:
@mickmister, maybe the structs tags are not correct and have `omitempty` when that are actually required?

Reply:
The `omitempty` is correct. But in all context's that `createBotPost` is called, there will be a `channel_id`.

Reply:
This code has been removed. The bot can post to any channel now and does not need to be added to a channel or team